components/infobox/wikis/chess/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,150 @@
+local Ratings = require('Module:Ratings')
+local Table = require('Module:Table')
+local TeamHistoryAuto = require('Module:TeamHistoryAuto')
+local Variables = require('Module:Variables')
+
+local Injector = Lua.import('Module:Widget/Injector')
+local Player = Lua.import('Module:Infobox/Person')
+
+local Widgets = require('Module:Widget/All')
+local Cell = Widgets.Cell
+local Title = Widgets.Title
+
+---@class ChessInfoboxPlayer: Person
+local CustomPlayer = Class.new(Player)
+local CustomInjector = Class.new(Injector)
+
+local TITLES = {
+	{code = "gm", name = "Grandmaster"},
+	{code = "im", name = "International Master"},
+	{code = "wgm", name = "Woman Grandmaster"},
+	{code = "fm", name = "FIDE Master"},
+	{code = "wim", name = "Woman International Master"},
+	{code = "cm", name = "Candidate Master"},
+	{code = "wfm", name = "Woman FIDE Master"},
+	{code = "wcm", name = "Woman Candidate Master"},
+}
+
+---@param frame Frame
+---@return Html
+function CustomPlayer.run(frame)
+	local player = CustomPlayer(frame)
+	player:setWidgetInjector(CustomInjector(player))
+
+	player.args.id = player.args.id or player.args.romanized_name or player.args.name
+
+	player.args.autoTeam = true
+	player.args.history = TeamHistoryAuto.results{
+		convertrole = true,
+		addlpdbdata = true
+	}
+
+	-- Title.
+	player.args.title = Array.find(
+		TITLES,
+		function (title, _)
+			return Logic.isNotEmpty(player.args['title_' .. title.code])
+		end
+	)
+
+	-- Rating IDs.
+	player.args.rating_id1 = player.args.rating_id or player.args.rating_id1 or player.args.fide
+	player.args.rating_ids = {}
+	for _, id, _ in Table.iter.pairsByPrefix(player.args, 'rating_id') do
+		table.insert(player.args.rating_ids, id)
+	end
+
+	if Table.isNotEmpty(player.args.rating_ids) then
+		-- Player ratings.
+		player.args.rating_classical = Ratings.getRecent(player.args.rating_ids, 'Classical')
+		player.args.rating_classical_peak = Ratings.getPeak(player.args.rating_ids, 'Classical')
+		player.args.rating_rapid = Ratings.getRecent(player.args.rating_ids, 'Rapid')
+		player.args.rating_rapid_peak = Ratings.getPeak(player.args.rating_ids, 'Rapid')
+		player.args.rating_blitz = Ratings.getRecent(player.args.rating_ids, 'Blitz')
+		player.args.rating_blitz_peak = Ratings.getPeak(player.args.rating_ids, 'Blitz')
+	end
+
+	if Logic.isNotEmpty(player.args.image) and Logic.isEmpty(player.args.caption) then
+		table.insert(player.warnings, 'Player photo is missing caption!')
+	end
+
+	return player:createInfobox()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	local args = self.caller.args
+
+	if id == 'custom' then
+		-- Titles.
+		local hasTitle = Logic.isNotEmpty(args.title)
+		if hasTitle then
+			Array.extendWith(widgets,
+				{Title{children = 'Titles'}},
+				Array.map(
+					TITLES,
+					function (title)
+						return Cell{name = title.name, content = {args['title_' .. title.code]}}
+					end
+				)
+			)
+		end
```

Comment:
fwiw imo rather push into achievements
and work on condensing it so the infobox doesn't get super long